bin/critterget.py

```diff
@@ -164,6 +174,181 @@ def apicall_with_response_code(uri, attribs=None):
+    :param session_key: string, auth token for Splunk
+    :return: a requests object
+    """
+    url = 'https://localhost:8089{}'.format(uri)
+    headers = {'Authorization': 'Splunk {}'.format(session_key)}
+    params = {OUTPUT_MODE: JSON}
+    splunk_response = None
+
+    if method == GET:
+        try:
+            splunk_response = requests.get(url,
+                                           params=params,
+                                           headers=headers,
+                                           verify=False)
+        except requests.exceptions.Timeout as error:
+            print 'Connection timeout. Splunk API returned an error code:', error
+        except requests.exceptions.ConnectionError as error:
+            print 'Connection error. Splunk API returned an error code:', error
+        except requests.exceptions.HTTPError as error:
+            print 'HTTP error. Splunk API returned an error code:', error
+        except requests.exceptions.RequestException as error:
+            print 'Splunk API retuned an error code:', error
+
+    elif method == POST:
+        try:
+            splunk_response = requests.post(url,
+                                            params=params,
+                                            headers=headers,
+                                            verify=False)
+        except requests.exceptions.Timeout as error:
+            print 'Connection timeout. Splunk API returned an error code:', error
+        except requests.exceptions.ConnectionError as error:
+            print 'Connection error. Splunk API returned an error code:', error
+        except requests.exceptions.HTTPError as error:
+            print 'HTTP error. Splunk API returned an error code:', error
+        except requests.exceptions.RequestException as error:
+            print 'Splunk API returned an error code:', error
+    else:
+        print (u'{} MessageType="ApteligentError" Error: Splunk API caller '
+               u'got a bad method: {}'.format(DATETIME_OF_RUN, method))
+
+    return splunk_response.json()
+
+
+def run_splunk_search(search_name, session_key):
+    """
+    Dispatches a saved search via Splunk's API and returns the results.
+
+    :param search_name: string, the name of the saved search
+    :param session_key: string, auth token for Splunk
+    :return:
+    """
+    saved_searches_uri = ('/servicesNS/admin/'
+                          'crittercism_integration/saved/searches')
+    search_jobs_uri = '/services/search/jobs/{}/results'
+    dispatch_url = None
+
+    all_saved_searches = splunk_api_call(saved_searches_uri, GET, session_key)
+
+    for entry in all_saved_searches[ENTRY]:
+        if entry[NAME] == search_name:
+            dispatch_url = entry[LINKS][DISPATCH]
+
+    if dispatch_url:
+        r = splunk_api_call(dispatch_url, POST, session_key)
+        sid = r[SID]
+        time.sleep(.25)  # This lets Splunk finish running the search
```

Comment:
This was the smallest possible delay that let Splunk finish running its search. I'm wondering now if a small loop would be better -- what do you think?